itests/eth_deploy_test.go

```diff
@@ -94,17 +96,25 @@ func TestDeployment(t *testing.T) {
 	hash := client.EVM().SubmitTransaction(ctx, &tx)
 	fmt.Println(hash)
 
+	mpoolTx, err := client.EthGetTransactionByHash(ctx, &hash)
+	require.NoError(t, err)
+
+	// require that the hashes are identical
+	require.Equal(t, hash, mpoolTx.Hash)
+
+	// these fields should be nil because the tx hasn't landed on chain.
+	require.Nil(t, mpoolTx.BlockNumber)
+	require.Nil(t, mpoolTx.BlockHash)
+	require.Nil(t, mpoolTx.TransactionIndex)
+
 	changes, err := client.EthGetFilterChanges(ctx, pendingFilter)
 	require.NoError(t, err)
 	require.Len(t, changes.Results, 1)
 	require.Equal(t, hash.String(), changes.Results[0])
 
-	time.Sleep(5 * time.Second)
-
 	var receipt *api.EthTxReceipt
-	for i := 0; i < 10000000000; i++ {
+	for i := 0; i < 20; i++ {
```

Comment:
Let's just actually `StateWaitMsg` here instead of this funkiness -- if we don't have the tooling to easily get the msgCid, slap a TODO on it for now (I think the work @geoff-vball  is doing will make this easy to do).

Reply:
We do need to check that the receipt is null while the transaction hasn't landed on chain -- it's part of the test's coverage.

Reply:
Sure, keep doing that every 500ms, but the cue to stop is StateWaitMsg returning (not "i waited 10 seconds").

Reply:
We still need a timeout for the test (should not let it run indefinitely). We could have a timed context with StateWaitMsg, but at that point we're achieving the same effect as here with more complicated code, so it's a matter of choice.